langchain4j/src/main/java/dev/langchain4j/memory/chat/ChatMemories.java

```diff
@@ -0,0 +1,40 @@
+package dev.langchain4j.memory.chat;
+
+import dev.langchain4j.memory.ChatMemory;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+public class ChatMemories {
+
+    public static final String DEFAULT = "default";
+
+    private ChatMemory defaultChatMemories;
+    private Map<Object, ChatMemory> chatMemories;
+    private ChatMemoryProvider chatMemoryProvider;
+
+    public ChatMemories(ChatMemoryProvider chatMemoryProvider) {
+        this.chatMemories = Collections.synchronizedMap(new WeakHashMap<>());
+        this.chatMemoryProvider = chatMemoryProvider;
```

Comment:
```suggestion
        this.chatMemoryProvider = ensureNotNull(chatMemoryProvider, "chatMemoryProvider");
```
Otherwise for `AiServices.builder(...).chatMemoryProvider(null)` case it will create `ChatMemories` instance and behave as if AI Service has chat memory (see `AiServiceContext.hasChatMemory()`)